pylint/checkers/classes/class_checker.py

```diff
@@ -1775,23 +1792,20 @@ def _is_type_self_call(self, expr: nodes.NodeNG) -> bool:
         )
 
     @staticmethod
-    def _is_classmethod(func):
+    def _is_classmethod(func: LocalsDictNodeNG) -> bool:
         """Check if the given *func* node is a class method."""
-
         return isinstance(func, nodes.FunctionDef) and (
             func.type == "classmethod" or func.name == "__class_getitem__"
         )
 
     @staticmethod
-    def _is_inferred_instance(expr, klass):
+    def _is_inferred_instance(expr: nodes.NodeNG | None, klass: nodes.ClassDef) -> bool:
```

Comment:
This would work but maybe we could add best effort values in pylint and see where mypy is unhappy once astroid is actually typed ? (The overload for ``infer`` depending on what node we're inferring and based on condition on the node can become quite complicated, and won't be done in a month).

Reply:
The issue is that we won't know which signatures to revisit when we finally do update `astroid`. For any typing for which we are not sure that `astroid` is correct I think it is often better to leave them be and try and fix `astroid` first.